packages/authentication/modules/access/usePreAuth/index.ts

```diff
@@ -0,0 +1,60 @@
+import { ACCESS_COOKIE_NAME, REFRESH_COOKIE_NAME, TokenTypes } from '@baseapp-frontend/utils'
+
+import { useQuery, useQueryClient } from '@tanstack/react-query'
+import Cookies from 'js-cookie'
+
+import AuthApi from '../../../services/auth'
+import { USER_API_KEY } from '../../../services/user'
+import { useSimpleTokenUser } from '../../user'
+import { IUsePreAuth } from './types'
+import { isJWTResponse } from './utils'
+
+const usePreAuth = ({
+  token,
+  queryOptions = {},
+  tokenType = TokenTypes.jwt,
+  cookieName = ACCESS_COOKIE_NAME,
+  refreshCookieName = REFRESH_COOKIE_NAME,
+  ApiClass = AuthApi,
+}: IUsePreAuth) => {
+  const queryClient = useQueryClient()
+  const { refetch: refetchUser } = useSimpleTokenUser({ options: { enabled: false } })
+
+  const query = useQuery({
+    staleTime: Infinity, // makes cache never expire automatically
+    enabled: !!token,
+    useErrorBoundary: false,
+    retry: false,
+    ...queryOptions, // needs to be placed bellow all overridable options
+    queryKey: ['preAuth', token, tokenType],
```

Comment:
can we set something similar to this in the `packages/authentication/services/auth.ts`?
```
export const AUTH_API_KEY = {
  default: ['auth'],
  preAuth: (token: string, tokenType: TokenTypes) => [...AUTH_API_KEY.default, 'preAuth', token, tokenType] as const,
}
```
and use it here instead? just so we can reuse that same query key builder in other places if needed